ouroboros-consensus/test-consensus/Test/Consensus/ChainSyncClient.hs

```diff
@@ -198,20 +202,22 @@ newtype ServerUpdates =
 --
 -- Note that updates that are scheduled before the slot at which we start
 -- syncing help generate different chains to start syncing from.
-runChainSync :: forall m.
+runChainSync :: forall m tip.
        ( MonadAsync m
        , MonadFork  m
        , MonadMask  m
        , MonadTimer m
        , MonadThrow (STM m)
+       , MonadSay   m
+       , tip ~ (Point (Header TestBlock), BlockNo)
```

Comment:
Any downside to replacing `tip` with `(Point (Header TestBlock), BlockNo)` and removing the type parameter?

Reply:
Downwards `tip` is used in two different independent places which could be substituted with different types and the code would compile, but if they don't agree the test would fail.
I think it's better to use `tip ~ ...` and when the `tip` will change modify it in one place without getting a false test failure.